pkg/services/vimmachine.go

```diff
@@ -266,6 +266,10 @@ func (v *VimMachineService) reconcileNetwork(ctx context.Context, vimMachineCtx
 			Address: addr,
 		})
 	}
+	machineAddresses = append(machineAddresses, clusterv1.MachineAddress{
+		Type:    clusterv1.MachineInternalDNS,
+		Address: vm.GetName(),
+	})
```

Comment:
Could we add a test?

Reply:
Addressed in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2529